scripts/process/as_svg.R

```diff
@@ -0,0 +1,266 @@
+#' @example 
+#' sub_attribute_text("hovertext('{{ID}}-{{dog}}' evt)", 
+#'    data.frame(ID = c('g','f'), dog = c('ralph','cindy')))
+#' 
+#' @return a character vector of length nrow(data)
+sub_attribute_text <- function(template, data){
+  sapply(seq_len(nrow(data)), FUN = function(j) {
+    whisker::whisker.render(template, data = data[j,])
+  }, USE.NAMES = FALSE)
+}
+
+#' sub text and/or define new svg attributes
+#' 
+#' uses `sub_attribute_text` to create attribute strings
+#' 
+#' @param dataframe data to use for evaluation
+#' @param attributes key value pairs that may or may not contain 
+#' mustache {{key}} syntax
+#' @return a data.frame with new columns for each name in `attributes`
+svg_sub_attributes <- function(dataframe, attributes = NULL){
+  
+  if (!is.null(attributes)){
+    # replace existing attributes w/ these user-specified ones
+    # evaluate "mustache" keys w/ geom_df
+    attr_names <- names(attributes)
+    
+    # if mustache keys exist, evaluate with the context of the dataframe variables:
+    attrs_subbed <- sapply(attr_names, FUN = function(x) {
+      sub_attribute_text(template = attributes[[x]], data = dataframe)
+    })
+    
+    dataframe[attr_names] <- attrs_subbed
+  }
+  
+  return(dataframe)
+}
+
+#' convert a simple feature collection into an svg path element's `d` attribute
+#' 
+#' @param .object a simple feature collection, currently limited to polygons
+#' @param xlim min and max x values in the plotting range in the coordinates of `sfc_object`
+#' @param ylim min and max y values in the plotting range in the coordinates of `sfc_object`
+#' @param \dots additional arguments sent to `plot`
+#' @param width the width of the output figure in inches
+#' @param height the height of the output figure in inches
+#' @param pointsize used in svgstring...fonts?
+#' 
+#' @return a data.frame with the same data as `.object`, 
+#' but with the new attribute(s) added, and the geometry removed
+sf_to_path <- function(.object, xlim, ylim, ..., width, height, pointsize = 12){
+  
+  stopifnot(!is.null(.object))
+  stopifnot(packageVersion('svglite') == '1.2.0.9003')
+  
+  sfc_object <- .object[["geometry"]]
+  data_in <- data.frame(.object) %>% select(-geometry)
+  data_in$feature_id <- seq_len(nrow(data_in))
+  
+  # hmmm...with sf, each polygon of a multi-poly plots as a separate `d`. 
+  # Can get this w/ st_cast(sfc_object[j], "POLYGON")
+  # could test for this `"sfc_MULTIPOLYGON" %in% class(sfc_object[j])`
+  # use `feature_id` to track which feature each polygon came from
+  feature_ids <- c()
+  rendered <- svglite::xmlSVG(width = width, height = height, pointsize = pointsize, standalone = F, {
+    
+    set_svg_plot(xlim, ylim)
+    for (j in seq_len(length(sfc_object))){
+      for (poly in sf::st_cast(sfc_object[j], "POLYGON")){
+        plot(poly, ..., add = TRUE)
+        feature_ids <- c(feature_ids, j)
+      }
+    }
+  })
+  
+  
+  paths <- xml_child(rendered) %>% xml_children
```

Comment:
where are these two xml functions from? I can't find them anywhere.

Reply:
I didn't have `xml2` installed and now do. You should add `xml2::` to be explicit.

Reply:
maybe...but just to offer a counter argument. I am not using a namespace for %>%, and also, those functions are all prefixed with `xml_`, so I thought it was redundant to add `xml2`

But I think you are right, we should stick to that practice.